collated/test/A0140022H.md

```diff
@@ -0,0 +1,1104 @@
+        DateTime dateTimeSource = new DateTime("14/01/2016 1200", "16/01/2016 1200");
+        DateTime dateTimeQueryFullyInRange = new DateTime("14/01/2016 2000", "15/01/2016 1200");
+        DateTime dateTimeQueryPartiallyInRange = new DateTime("13/01/2016 1000", "15/01/2016 1200");
+
+        assertTrue(DateTimeUtil.isDateTimeWithinRange(dateTimeSource, dateTimeQueryFullyInRange));
+        assertTrue(DateTimeUtil.isDateTimeWithinRange(dateTimeSource, dateTimeQueryPartiallyInRange));
+    }
+
+    @Test
+    public void isDateTimeWithinRange_dateTimeWithoutStartDate() throws DateTimeException, IllegalDateException {
+        DateTime dateTimeSource = new DateTime("15/01/2016 1200", "17/01/2016 1100");
+        DateTime dateTimeSourceWithoutStartDate = new DateTime("", "16/01/2016 1200");
+        DateTime dateTimeQuery = new DateTime("14/01/2016 2000", "17/01/2016 1200");
+        DateTime dateTimeQueryWithoutStartDate = new DateTime("", "16/01/2016 1200");
+        DateTime dateTimeQueryWithoutStartDate2 = new DateTime("", "18/01/2016 1200");
+
+        assertTrue(DateTimeUtil.isDateTimeWithinRange(dateTimeSource, dateTimeQuery));
+        assertFalse(DateTimeUtil.isDateTimeWithinRange(dateTimeSource, dateTimeQueryWithoutStartDate2));
+        assertTrue(DateTimeUtil.isDateTimeWithinRange(dateTimeSourceWithoutStartDate, dateTimeQuery));
+        assertTrue(DateTimeUtil.isDateTimeWithinRange(dateTimeSourceWithoutStartDate, dateTimeQueryWithoutStartDate));
+        assertFalse(DateTimeUtil.isDateTimeWithinRange(dateTimeSourceWithoutStartDate, dateTimeQueryWithoutStartDate2));
+    }
+
+}
+```
+###### \java\tars\logic\LogicManagerTest.java
+``` java
+     */
+    private void assertCommandBehaviorForList(String inputCommand, String expectedMessage, ReadOnlyTars expectedTars,
+            List<? extends ReadOnlyTask> expectedShownList) throws Exception {
+
+        // Execute the command
+        CommandResult result = logic.execute(inputCommand);
+
+        // Confirm the ui display elements should contain the right data
+        assertEquals(expectedMessage, result.feedbackToUser);
+        assertEquals(expectedShownList, model.getFilteredTaskList());
+
+        // Confirm the state of data (saved and in-memory) is as expected
+        assertEquals(expectedTars, latestSavedTars);
+    }
+
+    private void assertCommandBehaviorWithRsvTaskList(String inputCommand, String expectedMessage,
+            ReadOnlyTars expectedTars, List<? extends ReadOnlyTask> expectedShownTaskList,
+            List<? extends RsvTask> expectedShownRsvTaskList) throws Exception {
+
+        // Execute the command
+        CommandResult result = logic.execute(inputCommand);
+
+        // Confirm the ui display elements should contain the right data
+        assertEquals(expectedMessage, result.feedbackToUser);
+        assertEquals(expectedShownTaskList, model.getFilteredTaskList());
+        assertEquals(expectedShownRsvTaskList, model.getFilteredRsvTaskList());
```

Comment:
try to only hv 1 assert per test, tests have to be specific unless really necessary ~